packages/router/src/grpc/view-protocol-server/wallet-id.ts

```diff
@@ -1,22 +1,17 @@
-import {
-  WalletIdRequest,
-  WalletIdResponse,
-} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1alpha1/view_pb';
-import { localExtStorage } from '@penumbra-zone/storage';
-import { WalletId } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1alpha1/keys_pb';
-import { stringToUint8Array } from '@penumbra-zone/types';
-import { AnyMessage } from '@bufbuild/protobuf';
+import type { Impl } from '.';
+import { extLocalCtx } from '../../ctx';
 
-export const isWalletIdRequest = (req: AnyMessage): req is WalletIdRequest => {
-  return req.getType().typeName === WalletIdRequest.typeName;
-};
+import { stringToUint8Array } from '@penumbra-zone/types';
 
-export const handleWalletIdReq = async (): Promise<WalletIdResponse> => {
-  const wallets = await localExtStorage.get('wallets');
+import { WalletId } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1alpha1/keys_pb';
 
-  if (!wallets.length) return new WalletIdResponse();
+import { ConnectError, Code } from '@connectrpc/connect';
 
-  return new WalletIdResponse({
-    walletId: new WalletId({ inner: stringToUint8Array(wallets[0]!.id) }),
-  });
+// TODO: there is never data in this request. should a wallet index be available in context?
```

Comment:
Don't think there is such a thing as wallet index in the rpc defintion

Reply:
ok. there are a few places in the code that seem to require a wallet index to access storage, often defaulting to 0. seems like a separate issue